Index: src/components/form/FormControl.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/components/form/FormControl.tsx	(revision 9a61ed4233efedb92cf13fcf41ff5369cf8dc528)
+++ src/components/form/FormControl.tsx	(date 1554458212951)
@@ -1,9 +1,9 @@
-import React, { ComponentProps, ReactNode } from "react";
-import { FormGroup, Input, Label, Col, Row, NavLink } from "reactstrap";
-import { NavLink as Link } from "react-router-dom";
+import React, { ComponentProps, FC, ReactNode } from "react";
+import { FormGroup, Input } from "reactstrap";
 import { ErrorMessage, Field, FormikProps } from "formik";
 
 import "./FormControl.scss";
+import { FormControlLabel, FormControlLabelProps } from "./FormControlLabel";
 
 interface Props<
   TFormValues extends {},
@@ -12,11 +12,8 @@
   formikProps: TFormikProps;
   name: keyof TFormValues & string;
   label: ReactNode;
+  renderLabel?: FC<FormControlLabelProps>;
   type?: ComponentProps<typeof Input>["type"];
-  link?: {
-    label: string;
-    url: string;
-  } | null;
   placeholder?: string;
 }
 
@@ -24,23 +21,12 @@
   formikProps,
   name,
   label,
+  renderLabel = FormControlLabel,
   type = "text",
-  link = null,
   placeholder
 }: Props<TFormValues>) => (
   <FormGroup>
-    <Row>
-      <Col md={9}>
-        <Label for={`formControl_${name}`}>{label}</Label>
-      </Col>
-      {!!link && (
-        <Col md={3}>
-          <NavLink tag={Link} to={link.url} className="form-control__link">
-            {link.label}
-          </NavLink>
-        </Col>
-      )}
-    </Row>
+    {renderLabel({ name, label })}
     <Input
       type={type}
       name={name}
Index: src/components/form/FormControlLabel.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/components/form/FormControlLabel.tsx	(date 1554458153491)
+++ src/components/form/FormControlLabel.tsx	(date 1554458153491)
@@ -0,0 +1,12 @@
+import React, { FC, ReactNode } from "react";
+import { Label } from "reactstrap";
+
+export interface FormControlLabelProps {
+  name: string;
+  label: ReactNode;
+}
+
+export const FormControlLabel: FC<FormControlLabelProps> = ({
+  name,
+  label
+}) => <Label for={`formControl_${name}`}>{label}</Label>;
Index: src/app/features/auth/signin/SignInForm.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/features/auth/signin/SignInForm.tsx	(revision 9a61ed4233efedb92cf13fcf41ff5369cf8dc528)
+++ src/app/features/auth/signin/SignInForm.tsx	(date 1554458375735)
@@ -7,6 +7,8 @@
 
 import "./SignInForm.scss";
 import { ROUTES } from "../../routing/routes";
+import { Link } from "react-router-dom";
+import { FormControlLabel } from "../../../../components/form/FormControlLabel";
 
 export const SignInForm: FC<{
   onSubmit(d: SignInData): void;
@@ -33,13 +35,15 @@
         />
         <FormControl
           label="Password"
+          renderLabel={labelProps => (
+            <div style={{ display: "flex", justifyContent: "space-between" }}>
+              <FormControlLabel {...labelProps} />
+              <Link to={ROUTES.resetPassword}>Forgot your password?</Link>
+            </div>
+          )}
           name="password"
           type="password"
           formikProps={formikProps}
-          link={{
-            label: "Forgot your password?",
-            url: ROUTES.resetPassword
-          }}
           placeholder="******"
         />
         <FormGroup check inline>
